src/olympic.ml

```diff
@@ -11,17 +13,96 @@ let load_feature_file fname =
   let pickleLst = Gherkin.load_feature_file fname in
   List.rev_map (fun p -> {p with Cucumber.Pickle.steps = (List.rev p.Cucumber.Pickle.steps)}) pickleLst
 
-let unpack_pickle pickle =
-  pickle.Cucumber.Pickle.steps
-     
+let execute_step tests step =
+  try
+    let outcome = Cucumber.Lib.run tests step in
+    Ok outcome
+  with
+  | ex -> Error (Printexc.to_string ex)
+
+let execute_pickle tests pickle =
+  let name = pickle.Cucumber.Pickle.name in
+  let steps = pickle.Cucumber.Pickle.steps in
+
+  let (_, error, outcomeLst) = List.fold_left (fun (skipping, error, outcomes) step ->
+      if skipping then (true, error, Skip :: outcomes)
+      else
+        match execute_step tests step with
+        | Ok Pass -> (false, None, Pass :: outcomes)
+        | Ok o    -> (true, None, o :: outcomes)
+        | Error e -> (true, Some e, Fail :: outcomes)
+    ) (false, None, []) steps in
+
+  let _ = match error with
+    | Some e -> print_endline ("Error in scenario " ^ name);
+                print_endline e;
+                print_newline ()
+    | _ -> ()
+  in
+
+  List.rev outcomeLst
+
+let count_outcome outcome outcomeLst =
+  List.length (List.find_all (fun o -> o = outcome) outcomeLst)
+
+let rec print_list = function
+  | [] -> ()
+  | [x] -> print_string x;
+  | x::xs -> print_string x; print_string ", "; print_list xs
+
+let print_scenario_report outcomeLists =
+  let scenarios = List.length outcomeLists in
+
+  let failed = List.length (List.find_all (fun os ->
+      (count_outcome Fail os) > 0 || (count_outcome Pending os) > 0
+    ) outcomeLists) in
+
+  let undefined = List.length (List.find_all (fun os ->
+      (count_outcome Undefined os) > 0
+    ) outcomeLists) in
+
+  let passed = scenarios - failed - undefined in
+
+  let stats = [] in
+  let stats = if passed    = 0 then stats else (string_of_int passed ^ " passed")       :: stats in
```

Comment:
I started out like that but then I ended up on this uniform way of writing because I find this much easier to read: it shows the structural similarity of these assignments and highlights the differences.

But sure, I will change it to your preferred style.